apollo-runtime/api/apollo-runtime.api

```diff
@@ -182,42 +187,53 @@ public final class com/apollographql/apollo3/network/OkHttpExtensionsKt {
-	public static final field Companion Lcom/apollographql/apollo3/network/http/BatchingHttpEngine$Companion;
 	public fun <init> ()V
 	public fun <init> (Lcom/apollographql/apollo3/network/http/HttpEngine;)V
 	public fun <init> (Lcom/apollographql/apollo3/network/http/HttpEngine;J)V
 	public fun <init> (Lcom/apollographql/apollo3/network/http/HttpEngine;JI)V
 	public synthetic fun <init> (Lcom/apollographql/apollo3/network/http/HttpEngine;JIILkotlin/jvm/internal/DefaultConstructorMarker;)V
+	public fun dispose ()V
+	public fun execute (Lcom/apollographql/apollo3/api/http/HttpRequest;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public final fun getDelegate ()Lcom/apollographql/apollo3/network/http/HttpEngine;
+}
+
+public final class com/apollographql/apollo3/network/http/BatchingHttpInterceptor : com/apollographql/apollo3/network/http/HttpInterceptor {
+	public static final field CAN_BE_BATCHED Ljava/lang/String;
+	public static final field Companion Lcom/apollographql/apollo3/network/http/BatchingHttpInterceptor$Companion;
+	public fun <init> ()V
+	public fun <init> (J)V
+	public fun <init> (JI)V
+	public synthetic fun <init> (JIILkotlin/jvm/internal/DefaultConstructorMarker;)V
 	public static final fun configureApolloCall (Lcom/apollographql/apollo3/ApolloCall;Z)V
 	public static final fun configureApolloClientBuilder (Lcom/apollographql/apollo3/ApolloClient$Builder;Z)V
 	public fun dispose ()V
-	public fun execute (Lcom/apollographql/apollo3/api/http/HttpRequest;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 	public final fun getBatchIntervalMillis ()J
-	public final fun getDelegate ()Lcom/apollographql/apollo3/network/http/HttpEngine;
+	public fun intercept (Lcom/apollographql/apollo3/api/http/HttpRequest;Lcom/apollographql/apollo3/network/http/HttpInterceptorChain;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 }
 
-public final class com/apollographql/apollo3/network/http/BatchingHttpEngine$Companion {
+public final class com/apollographql/apollo3/network/http/BatchingHttpInterceptor$Companion {
 	public final fun configureApolloCall (Lcom/apollographql/apollo3/ApolloCall;Z)V
 	public final fun configureApolloClientBuilder (Lcom/apollographql/apollo3/ApolloClient$Builder;Z)V
 }
 
-public final class com/apollographql/apollo3/network/http/BatchingHttpEngine$PendingRequest {
+public final class com/apollographql/apollo3/network/http/BatchingHttpInterceptor$PendingRequest {
 	public fun <init> (Lcom/apollographql/apollo3/api/http/HttpRequest;)V
 	public final fun getDeferred ()Lkotlinx/coroutines/CompletableDeferred;
 	public final fun getRequest ()Lcom/apollographql/apollo3/api/http/HttpRequest;
 }
 
-public final class com/apollographql/apollo3/network/http/BatchingHttpEngineExtensions {
+public final class com/apollographql/apollo3/network/http/BatchingHttpInterceptorKt {
 	public static final fun canBeBatched (Lcom/apollographql/apollo3/api/MutableExecutionOptions;Z)Ljava/lang/Object;
```

Comment:
Looks like this one can move to `MutableExecutionOptions` instead of being an extension function. Want to look into it?

Reply:
Yes :) 👀